src/server/server.js

```diff
@@ -1,191 +1,33 @@
-
-    var newUser = {};
-    newUser.email = req.body.email.toLowerCase();
-    newUser.username = req.body.username.trim();
-    newUser.password = req.body.password;
-
-    User.create(newUser, function (err, user) {
-        if (err) {
-            req.flash('error', err.message);
-            return res.redirect('back');
-        } else {
-            // Passport exposes a login() function on req (also aliased as logIn())
-            // that can be used to establish a login session. This function is
-            // primarily used when users sign up, during which req.login() can
-            // be invoked to log in the newly registered user.
-            req.login(user, function (err) {
-                if (err) {
-                    req.flash('error', err.message);
-                    return res.redirect('back');
-                }
-                return res.redirect('/auth/account');
-            });
-        }
-    });
-});
-
-app.get('/login', function (req, res, next) {
-    res.render('pages/login', {
-        user: req.user,
-        url: req.url
-    });
-});
-
-app.get('/auth/logout', function (req, res, next) {
-    rpeq.logout();
-    res.redirect('/');
-});
+var PassportConfigurator =
+  require('loopback-component-passport').PassportConfigurator;
 
-// -- Mount static files here--
-// All static middleware should be registered at the end, as all requests
-// passing the static middleware are hitting the file system
-// Example:
-var path = require('path');
-app.use(loopback.static(path.resolve(__dirname, '../client/public')));
-
-// Requests that get this far won't be handled
-// by any middleware. Convert them into a 404 error
-// that will be handled later down the chain.
-//app.use(loopback.urlNotFound());
-
-// The ultimate error handler.
-app.use(loopback.errorHandler());
+var app = module.exports = loopback();
 
-app.start = function () {
-    // start the web server
-    return app.listen(function () {
-        app.emit('started');
-        var baseUrl = app.get('url').replace(/\/$/, '');
-        console.log('%s %s', __dirname, util.inspect(app.get('url')));
-        console.log('Web server listening at: %s', baseUrl);
-        if (app.get('loopback-component-explorer')) {
-            var explorerPath = app.get('loopback-component-explorer').mountPath;
-            console.log('Browse your REST API at %s%s', baseUrl, explorerPath);
-        }
-    });
+app.start = function() {
+  // start the web server
+  return app.listen(function() {
+    app.emit('started');
+    console.log('Web server listening at: %s', app.get('url'));
+    app.emit('started');
+    var baseUrl = app.get('url').replace(/\/$/, '');
+    console.log('Web server listening at: %s', baseUrl);
+    if (app.get('loopback-component-explorer')) {
+      var explorerPath = app.get('loopback-component-explorer').mountPath;
+      console.log('Browse your REST API at %s%s', baseUrl, explorerPath);
+    }
+  });
 };
 
-app.boot = function (cb) {
```

Comment:
app.boot function has been created so that it can be called from test integration tests. Can we keep this structure just for app.boot? I am ok with all other changes in this file.